rainier-core/src/main/scala/com/stripe/rainier/compute/Target.scala

```diff
@@ -18,6 +18,45 @@ class Target(val real: Real, val placeholders: Map[Variable, Array[Double]]) {
 
   private def nRows(data: Map[Variable, Array[Double]]): Int =
     data.head._2.size
+
+  def batched(nBatches: Int): (Target, Real) =
+    if (placeholders.isEmpty)
+      (this, Real.zero)
+    else if (nBatches == 1)
+      (Target.empty, inlined)
+    else {
+      val n = nRows(placeholders)
+      val variables = placeholders.keys.toList
+      val nColumns = n / nBatches
+      val copies = 0.until(nColumns).map { i =>
+        val columnVariables = variables.map { v =>
+          v -> new Variable
+        }.toMap
+        val columnReal = PartialEvaluator.inline(real, columnVariables)
+        (i, columnReal, columnVariables)
+      }
+      val newReal = Real.sum(copies.map(_._2))
```

Comment:
I understand the "make a copy of the original `real` with the cloned variables substituted for the original ones" but don't totally get the "and them sum these to make the `newReal`." I get that we want a single `newReal` at the end and don't _really_ see another way to make one, but why is the sum what we want?

Reply:
That's just the semantics of `Target`. We use it when we have a log-likelihood function `f` and a list of observations `x`, where the non-batchable version would be `x.map(f).sum`.